apps/prairielearn/src/pages/administratorQuery/administratorQuery.ts

```diff
@@ -98,44 +91,44 @@ router.post(
   '/:query',
   asyncHandler(async (req, res, _next) => {
     const jsonFilename = req.params.query + '.json';
-    const sqlFilename = req.params.query + '.sql';
+    const jsFilename = req.params.query + '.js';
 
-    const info = await jsonLoad.readJSON(path.join(queriesDir, jsonFilename));
-    const querySql = await fs.readFile(path.join(queriesDir, sqlFilename), {
-      encoding: 'utf8',
-    });
+    const info = AdministratorQuerySchema.parse(
+      await jsonLoad.readJSON(path.join(queriesDir, jsonFilename)),
+    );
 
-    const queryParams = {};
-    if (info.params) {
-      info.params.forEach((p) => {
-        queryParams[p.name] = req.body[p.name];
-      });
-    }
+    const queryParams: Record<string, string> = {};
+    info.params?.forEach((p) => {
+      queryParams[p.name] = req.body[p.name];
+    });
 
-    const params: AdministratorQueryRunParams = {
-      name: req.params.query,
-      sql: querySql,
-      params: queryParams,
-      authn_user_id: res.locals.authn_user.user_id,
-      error: null,
-      result: null,
-    };
+    let sqlOrModuleInfo = '';
+    let error: string | null = null;
+    let result: AdministratorQueryResult | null = null;
     try {
-      const result = await sqldb.queryAsync(querySql, queryParams);
-      params.result = {
-        rowCount: result.rowCount ?? 0,
-        columns: result.fields.map((f) => f.name),
-        rows: result.rows,
-      };
+      const module = await import(path.join(queriesDir, jsFilename));
+      sqlOrModuleInfo = module.default.toString();
+      result = (await module.default(queryParams)) as AdministratorQueryResult;
     } catch (err) {
-      params.error = err.toString();
+      logger.error(err);
+      error = err.toString();
     }
-    const result = await sqldb.queryRow(
+
+    const query_run_id = await sqldb.queryRow(
       sql.insert_query_run,
-      params,
-      AdministratorQueryQueryRunIDSchema,
+      {
+        name: req.params.query,
+        sql: sqlOrModuleInfo,
```

Comment:
AFAICT we never read this column anymore. Is that right, and if so, will you remove it after this PR is deployed?

Reply:
I wasn't sure if this would be valuable in the future, so I didn't move in that direction yet, but I'm open to removing it in the future. I could remove all references to the column now in this PR, and add a migration to make it nullable, to make that transition easier in a future PR, do you want me to do that?

Reply:
That might be good. I think there's little value in saving a stringified version of a function to the database.